src/Nethermind/Nethermind.Merge.AuRa/AuRaMergePlugin.cs

```diff
@@ -73,10 +84,81 @@ protected override PostMergeBlockProducerFactory CreateBlockProducerFactory()
                 _blocksConfig,
                 _api.LogManager);
 
+        protected override BlockProducerEnv CreateBlockProducerEnv()
+        {
+            Debug.Assert(_api?.BlockProducerEnvFactory is not null,
+                $"{nameof(_api.BlockProducerEnvFactory)} has not been initialized.");
+
+            Logging.ILogger logger = _api.LogManager.GetClassLogger();
+
+            ShutterTxSource? shutterTxSource = null;
+
+            if (_auraConfig!.UseShutter)
+            {
+                // parse validator info file (index, pk)
+                Dictionary<ulong, byte[]> validatorsInfo = [];
+                try
+                {
+                    // todo: parse as normal class?
+                    JsonDocument validatorsInfoDoc = JsonDocument.Parse(File.ReadAllText(_auraConfig.ShutterValidatorInfoFile));
+                    validatorsInfo = validatorsInfoDoc.RootElement.EnumerateObject().ToDictionary((JsonProperty p) => Convert.ToUInt64(p.Name), (JsonProperty p) => Convert.FromHexString(p.Value.GetString()!.AsSpan(2)));
+                }
+                catch (FileNotFoundException e)
+                {
+                    throw new FileNotFoundException($"Could not find Shutter validator info file: {e}");
+                }
+                catch (DirectoryNotFoundException e)
+                {
+                    throw new DirectoryNotFoundException($"Could not find Shutter validator info file: {e}");
+                }
+                catch (UnauthorizedAccessException e)
+                {
+                    throw new UnauthorizedAccessException($"Could not access Shutter validator info file (update permissions): {e}");
+                }
+                catch (IOException e)
+                {
+                    throw new IOException($"Could not load Shutter validator info file: {e}");
+                }
+                catch (JsonException e)
+                {
+                    throw new JsonException($"Could not parse Shutter validator info file: {e}");
+                }
```

Comment:
There is no problem in catching `System.Exception` if you intend rethrow it. Just remember to pass the original exception as arg.